### DISCUSSION TITLE
1 number 2 webhooks = (#100) Invalid parameter

### DISCUSSION THREAD

**Discussion by xenastro** [gabrieldwight/Whatsapp-Business-Cloud-Api-Net#86] [Q&A] [answered]
Hello,

I have an application that is live on production.

I was trying to write new code that would send welcome messages, so I wrote the code and wanted to test it out on my local machine before releasing it to production, but did not want to change the webhook to point to my local machine (with the help of Ngrok to tunnel it), so I created another test app on Facebook developer console and set the Ngrok URL that points to my local machine in the webhook configuration of the new app.

The new app is under the same Facebook business account, so now the live WhatsApp phone number and the test phone numbers are both attached to multiple Facebook Applications and we ended up with two webhooks.

I tried sending a message to one of the numbers, and I received multiple webhook requests to both endpoints (live and local) and in my code I use your API to mark the message as read, when I do that I get this error:

`WhatsappBusiness.CloudApi.Exceptions.WhatsappBusinessCloudAPIException: (#100) Invalid parameter     at WhatsappBusiness.CloudApi.WhatsAppBusinessClient.WhatsAppBusinessPostAsync[T](Object whatsAppDto, String whatsAppBusinessEndpoint, CancellationToken cancellationToken)     at WhatsappBusiness.CloudApi.WhatsAppBusinessClient.MarkMessageAsReadAsync(MarkMessageRequest markMessage, WhatsAppBusinessCloudApiConfig cloudApiConfig, CancellationToken cancellationToken)`

It does not specify which parameter is invalid, If I turn one of the webhooks off, I no longer get this error.

How are we supposed to test webhooks when our application is live? the only way I found out is to replace the webhook URL, but if the app is live all traffic will go to the new link and I will lose some messages, I try to do this in off hours but still might miss some requests.

Cheers.

**Comment by gabrieldwight** [comment, marked as answer]
Hi, on whatsapp you can only subscribe to one webhook endpoint for your applications as I can remember the option I had is to replace the webhook url. The phone number id might have a mismatch. [Error Codes](https://developers.facebook.com/docs/whatsapp/cloud-api/support/error-codes/)

**Reply by xenastro** [reply]
You are right, I tried to send the message from a number different than the number the message was received on.

But my solution actually worked, I now have two webhooks connected to the same phone number, one webhook pointing to my local machine and the other to the production, so I can test without losing messages, hurray!

Thank you again, Gabriel.

**Comment by fatihtas** [comment]
meta must provide 2 webhook endpoints. one for development, and another fpor production. this is ridiculous.

**Reply by xenastro** [reply]
Agreed.
